test/generators/mailbox_generator_test.rb

```diff
@@ -17,12 +17,6 @@ def test_mailbox_skeleton_is_created
       assert_match(/def process/, mailbox)
       assert_no_match(%r{# routing /something/i => :somewhere}, mailbox)
     end
-
-    assert_file "app/mailboxes/application_mailbox.rb" do |mailbox|
-      assert_match(/class ApplicationMailbox < ActionMailbox::Base/, mailbox)
-      assert_match(%r{# routing /something/i => :somewhere}, mailbox)
-      assert_no_match(/def process/, mailbox)
-    end
   end
 
```

Comment:
Instead of just removing this failing sections, I'd add a new spec to test the branch creating the `ApplicationMailbox`. Something like this: 

```suggestion
  def test_application_mailbox_is_created
    Object.send :remove_const, :ApplicationMailbox
    run_generator ['application']
    assert_file "app/mailboxes/application_mailbox.rb" do |mailbox|
      assert_match(/class ApplicationMailbox < ActionMailbox::Base/, mailbox)
      assert_match(%r{# routing /something/i => :somewhere}, mailbox)
      assert_no_match(/def process/, mailbox)
    end
  end

      end
```

PS: by the way, test is red without `Object.send :remove_const, :ApplicationMailbox`, due to class name collision. When invoked with `bin/rails action_mailbox:install` it works just fine. If someone can find a better way, happy to 👁.

Reply:
I tried exactly that way locally, but test failed. Haven't found constant removing, tahnk you 🙇